test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +93,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(command.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
```

Comment:
it would be good to check if the command that is NOT mutating indeed doesn't save the file

Reply:
I should have checked for that condition. Noted with thanks.